virtual/environment.go

```diff
@@ -484,7 +484,25 @@ func (r *environment) invokeActorStreamHelper(
 			"ensureActivation() success with 0 references for actor ID: %s", actorID)
 	}
 
-	return r.invokeReferences(ctx, vs, references, operation, payload, create)
+	idx, resp, err := r.invokeReferences(ctx, vs, references, operation, payload, create)
+	// If there is an error and it's not due to server blacklisting, retry the invocation.
+	// If the error is due to blacklisting, let the error propagate as it will be handled
+	// by the main InvokeActor function.
+	if err != nil && !IsBlacklistedActivationError(err) {
+		switch policy := create.Options.RetryPolicy; policy {
```

Comment:
Can you just do: `switch create.Options.RetryPolicy`? I dont see you using the extra variable anywhere anyways

Reply:
It is a good idea I think to update the error message to say whether a retry was performed or not. I.E if no retry was performed wrap the error to say that (and why) and if it was retried say how many attempts we made and that they all failed etc